### PR TITLE
Add RELEASING.md

### PR DESCRIPTION
**Pull request by reta**
### Description
Add RELEASING.md

### Issues Resolved
Follow up on https://github.com/opensearch-project/opensearch-testcontainers/pull/114#issuecomment-1819835888

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@gaiksaya @peterzhuamazon please, adding RELEASING.md, thank you

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+## Release Labels
+
+Repositories create consistent release labels, such as `v1.0.0`, `v1.1.0` and `v2.0.0`, as well as `patch` and `backport`. Use release labels to target an issue or a PR for a given release.
+
+## Releasing
+
+The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [maintainers](MAINTAINERS.md).
+
+1. Create a tag, e.g. 1.0.0, and push it to this GitHub repository.
+1. The [release-drafter.yml](.github/workflows/release-drafter.yml) will be automatically kicked off and a draft release will be created.
```
Comment (gaiksaya) [inline RELEASING.md:38]: Maybe add the approval process too?


```
1. Before creating a draft release, this workflow creates a GitHub issue asking for approval from the [maintainers](MAINTAINERS.md). See sample [issue](https://github.com/opensearch-project/opensearch-java/issues/733). The maintainers need to approve in order to continue the workflow run.
```

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+1. This draft release triggers the [jenkins release workflow](https://build.ci.opensearch.org/job/opensearch-testcontainers-release) as a result of which the client is released on [maven central](https://search.maven.org/search?q=org.opensearch). Please note that the release workflow is triggered only if created release is in draft state.
```
Comment (gaiksaya) [inline RELEASING.md:39]:
```suggestion
1. This draft release triggers the [jenkins release workflow](https://build.ci.opensearch.org/job/opensearch-testcontainers-release) as a result of which the client is released on [maven central](https://central.sonatype.com/artifact/org.opensearch/opensearch-testcontainers). Please note that the release workflow is triggered only if created release is in draft state.
```

File: RELEASING.md (hunk 1 above)
Comment (reta) [reply RELEASING.md]: Updated, thanks @gaiksaya !

**Review by gaiksaya — APPROVED** [review]
Looks good overall. Minor nits.

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+Repositories create consistent release labels, such as `v1.0.0`, `v1.1.0` and `v2.0.0`, as well as `patch` and `backport`. Use release labels to target an issue or a PR for a given release.
+
+## Releasing
+
+The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [maintainers](MAINTAINERS.md).
+
+1. Create a tag, e.g. 1.0.0, and push it to this GitHub repository.
+1. The [release-drafter.yml](.github/workflows/release-drafter.yml) will be automatically kicked off. This workflow creates a GitHub issue asking for approval from the [maintainers](MAINTAINERS.md) (see sample [issue](https://github.com/opensearch-project/opensearch-java/issues/733)). The maintainers need to approve in order to continue the workflow run and draft release creation.
+
+1. This draft release triggers the [jenkins release workflow](https://build.ci.opensearch.org/job/opensearch-testcontainers-release) as a result of which the client is released on [maven central](https://central.sonatype.com/artifact/org.opensearch/opensearch-testcontainers). Please note that the release workflow is triggered only if created release is in draft state.
```
[the remark this answers is not in the document]
Comment (reta) [reply RELEASING.md:40]: :+1:

File: RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+1. Once the above release workflow is successful, the drafted release on GitHub is published automatically.
+1. Increment "version" in [version.properties](./version.properties) to the next iteration, e.g. v2.1.1. See [example](https://github.com/opensearch-project/opensearch-testcontainers/pull/39).
+
+## Snapshot Builds
+The [snapshots builds](https://aws.oss.sonatype.org/content/repositories/snapshots/org/opensearch/opensearch-testcontainers/) are published to sonatype using [publish-snapshots.yml](./.github/workflows/publish-snapshots.yml) workflow. Each `push` event to the main branch triggers this workflow.
```
[the remark this answers is not in the document]
Comment (reta) [reply RELEASING.md]: Addressed, thank you!

**Comment by reta** [comment]
@gaiksaya mind please taking one more look? thank you.

**Review by gaiksaya — APPROVED** [review]
Didn't realize review was discarded. LGTM!

**Pull request merged**